specs/modal/readme.md

```diff
@@ -0,0 +1,164 @@
+| onAfterOpen | func | undefined |  |  |
+| onRequestClose(source) | func | undefined |  | The function used to handle cancel events on the modal; typically sets the open state to false. It is passed a source, as one of a number of possible closing options |
+| children | any |  |  | children nodes of the modal |
+| width | number/enum | 'medium' |  | Explicitly set a numeric width or provide one of three sizes; 'small', 'medium', 'large' - 320px, 640px, 960px respectively|
+| height | number/enum | 'auto' |  | Explicitly set a numeric height of the modal|
+| container | any | null |  |  |
+| aria-* | - | - |  |  |
+| aria-role | - | 'dialog' |  |  |
+
+#### Modal Roles
+
+Modal children can have a `role` attribute that allows them to declare which part of the modal is described by the node.
+
+| role | class | description |
+| --- | --- | --- |
+| header | .header | the header of the modal, in many cases containing a title for the modal as well as a close button |
+| header-close-button | .header-close-button | the close button located in the header. Normally, at the top-right corner of the modal |
+| header-title | .header-title | the title of the modal found in the header |
+| body | .header-body | the body of the modal |
+| footer | .footer | the footer of the modal, usually containing at least two options: 1) to close or cancel the action, 2) to perform the primary action required by the modal (e.g., submit the form, login to the app) |
+| footer-close-button | .footer-close-button | the close or cancel button found in the footer |
+| footer-primary-button | .footer-primary-button | the primary button of the modal |
+
+Children with a role may also accept different props, such as a `title` for the child marked as header.
+
+### `onRequestClose` Sources
+
+| Source | Description |
+| --- | --- |
+| backdropClick | request to close the modal because the backdrop of the modal was clicked |
+| escKeyPress | request to close the modal because the `ESC` button was pressed |
+| headerCloseClick | request to close the modal came from a child in the role `header-close-button` |
+| footerCloseClick | request to close the modal came from a child in the role `footer-close-button` |
+| footerPrimaryClick | request to close the modal came from a child in the role `footer-primary-button`  |
+
+### Modal Header Props
+
+| Prop | Type | Default | Description |
+| --- | --- | --- | --- |
+| children | node | undefined | Alternative to using the text attribute, for when you need more control over the content|
+| onClose | func | undefined | What to do when the user clicks the close button|
+| title | string | '' | Creates a title for the modal |
+
+### Code Example
+
+**Example 1:**
+
+```jsx
+<Modal
+    isOpen={this.state.open}
+    onRequestClose={this.whenModalCloses}>
+    <div role={header}>
+        <span role={title}>Terms of service</span>
+        <button role={header-close-button} />
```

Comment:
`role={header-close-button}` should be `role="header-close-button"`. See also other cases below. `header-close-button` is not valid JavaScript/JSX symbol.